WordPress/gutenberg
# Remove `template-only` mode from editor and edit-post packages

## What?
Removes the `template-only` editor rendering mode.

## Why?
This mode was only added as a temporary measure to help with some of the work around integrating the site and post editors. With the work that was done in [57036](https://github.com/WordPress/gutenberg/pull/57036) this can now be removed as it is now possible to inject a new post into the editor provider, so the current post can be replaced with the template.

## How?
Uses the new `getPostLinkProps` editor setting to inject the template id and post type into the editor provider.

## Testing Instructions

- Add a new post and select the `Template - Edit Template` option in the right post panel
- Check that you navigate to the template and can edit it
- Use the back button in the top editor and make sure you go back to the post
- Check that when you click the post Save/Update button the template changes show as needing saving
- Also try the `Template - Create new template` option in right post panel and check it works
- Try the `Template - Edit template` option again and this time make sure the `Go back` option in the success snackbar works
- In the site editor open a page and repeat all of the above actions. The only difference will be that the `Go back` button will not appear in the notification, this is because the routing works differently in the site editor, and the top back link is more prominant
